near/omni-bridge/src/lib.rs

```diff
@@ -564,7 +588,11 @@ impl Contract {
         );
 
         let message = self.remove_transfer_message(fin_transfer.transfer_id);
-        let fee = message.amount.0 - fin_transfer.amount.0;
+        let de_normalized_amount = Self::de_normalize_amount(
+            fin_transfer.amount.0,
+            self.token_decimals.get(&message.token),
```

Comment:
We need tests to verify this pipeline. I suspect that message.token is the address of the original sender. So, in the case of a transfer from NEAR to Solana, it will be the address on NEAR. Consequently, the decimals will correspond to the NEAR address. However, to reconstruct how much was transferred, we would need the decimals from Solana.